### PR TITLE
Feature : Draft support

### PR DESCRIPTION
**Pull request by ganfra**
## Type of change

- [X] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content
Implement draft support in the following way : 
- When message screen is backgrounded, then the current draft is saved. It also keeps the composer mode (either Regular, Reply or Edit).
- When message screen is first composed we load the draft and delete it from the store. 

This PR also introduce some refactoring around ReplyTo, so we can use the same UI logic between Timeline item and Composer. 

## Motivation and context
Closes https://github.com/element-hq/element-x-android/issues/2869

## Tests
- Open a room
- Write a message without sending it
- Close the room
- Open back the room 
- You should see your message in the composer
- You can repeat the steps with text formatting enabled, and reply/edit modes.

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/messagecomposer/MessageComposerPresenter.kt [hunk 1]
```diff
@@ -570,4 +536,111 @@ class MessageComposerPresenter @Inject constructor(
...
+            draftService.saveDraft(room.roomId, composerDraft)
+        }
+    }
+
+    private fun CoroutineScope.toggleTextFormatting(
+        enabled: Boolean,
+        markdownTextEditorState: MarkdownTextEditorState,
+        richTextEditorState: RichTextEditorState,
+    ) = launch {
+        showTextFormatting = enabled
```
Comment (jmartinesp) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/messagecomposer/MessageComposerPresenter.kt:602]: Is this indended?

File: libraries/textcomposer/impl/src/main/kotlin/io/element/android/libraries/textcomposer/model/TextEditorState.kt [hunk 2]
```diff
@@ -45,6 +45,20 @@ sealed interface TextEditorState {
...
+        }
+    }
+
+    suspend fun setMarkdown(text: String) {
+        when (this) {
+            is Markdown -> state.text.update(text, true)
+            is Rich -> richTextEditorState.setMarkdown(text)
+        }
+    }
+
```
Comment (jmartinesp) [inline libraries/textcomposer/impl/src/main/kotlin/io/element/android/libraries/textcomposer/model/TextEditorState.kt:61]: Isn't this already included in the latest `develop`? Maybe the PR needs to merge the latest changes?

File: tests/uitests/src/test/snapshots/images/ui_S_t[l.textcomposer_TextComposerReply_null_TextComposerReply-Day-4_5_null_0,NEXUS_5,1.0,en].png
Comment (jmartinesp) [inline tests/uitests/src/test/snapshots/images/ui_S_t[l.textcomposer_TextComposerReply_null_TextComposerReply-Day-4_5_null_0,NEXUS_5,1.0,en].png:1]: That's a long composer! I guess the state is expanded?

**Review by jmartinesp — APPROVED** [review]
LGTM!

File: tests/uitests/src/test/snapshots/images/ui_S_t[l.textcomposer_TextComposerReply_null_TextComposerReply-Day-4_5_null_0,NEXUS_5,1.0,en].png
Comment (ganfra) [reply tests/uitests/src/test/snapshots/images/ui_S_t[l.textcomposer_TextComposerReply_null_TextComposerReply-Day-4_5_null_0,NEXUS_5,1.0,en].png:1]: I'm not sure why :/

Comment (frebib) [reply tests/uitests/src/test/snapshots/images/ui_S_t[l.textcomposer_TextComposerReply_null_TextComposerReply-Day-4_5_null_0,NEXUS_5,1.0,en].png:1]: Looks like https://github.com/element-hq/element-x-android/issues/2167

**Pull request merged**